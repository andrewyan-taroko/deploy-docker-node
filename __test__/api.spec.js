const { times }     = require('lodash');
const supertest     = require('supertest');

const server        = require('server.js');
const faker         = require('utils/faker.js');
const { sequelize } = require('models');

const request        = supertest(server);
const queryInterface = sequelize.getQueryInterface();

describe('testing API', () => {
  beforeAll(async () => {
    await sequelize.sync({ force: true });
    const users = times(10, (id) => ({
      id,
      name: faker.name.findName(),
      createdAt: new Date(),
      updatedAt: new Date(),
    }));
    queryInterface.bulkInsert("Users", users);
  });

  afterAll(async () => {
    queryInterface.bulkDelete("Users", null, {})
    await sequelize.close();
    server.close();
  });

  describe('/api', () => {
    it('should return fixed data', async () => {
      await request
        .get('/api')
        .expect(200)
        .then((res) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });
});
