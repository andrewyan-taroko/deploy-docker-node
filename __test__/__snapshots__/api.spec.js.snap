// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing API /api should return fixed data 1`] = `
Object {
  "header": Object {
    "connection": "close",
    "content-length": "1118",
    "content-type": "application/json; charset=utf-8",
    "date": "Tue, 13 Feb 2018 09:51:23 GMT",
    "etag": "W/\\"45e-nS7dyeD200mamhlh5+BovPx0GcY\\"",
    "x-powered-by": "Express",
  },
  "req": Object {
    "data": undefined,
    "headers": Object {
      "user-agent": "node-superagent/3.8.2",
    },
    "method": "GET",
    "url": "http://127.0.0.1:33781/api",
  },
  "status": 200,
  "text": "[{\\"id\\":0,\\"name\\":\\"Lilly Schumm III\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":1,\\"name\\":\\"Olen Satterfield\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":2,\\"name\\":\\"Camilla Gorczany\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":3,\\"name\\":\\"Raul Williamson Sr.\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":4,\\"name\\":\\"Florencio Ankunding\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":5,\\"name\\":\\"Marjory Harber\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":6,\\"name\\":\\"Furman Boyle\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":7,\\"name\\":\\"Jo Kiehn\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":8,\\"name\\":\\"Dr. Oliver Green\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"},{\\"id\\":9,\\"name\\":\\"Myron Morar\\",\\"createdAt\\":\\"2018-02-13T09:51:23.136Z\\",\\"updatedAt\\":\\"2018-02-13T09:51:23.136Z\\"}]",
}
`;
